src/common/config.cc

```diff
@@ -1236,21 +1236,45 @@ bool md_config_t::expand_meta(std::string &origval,
 }
 
 void md_config_t::diff(
+  const md_config_t *other,
+  map<string, pair<string, string> > *diff,
+  set<string> *unknown) 
```

Comment:
why do we need this method as we have `diff_helper()` whose last param is optional?